### PR TITLE
fix: [CDS-80666]: large console logs not rendering properly

### PR DESCRIPTION
**Pull request by vikrant-harness**
### Summary

 We debugged this issue under different conditions :-   
1. Presence of rich text  vs a normal text of same height at the start index (No difference in both , Issue is still present)  
2. Rendering the text in a normal div instead of our custom elements (No difference in both , Issue is still present)  
3. Checked the profilers when the scroll is happening (buggy scroll). (Couldn't find any frame drop in the profiler)  
4. Presence of the log line with height greater than the view port height being present at different positions  

Observation :- the presence of the line at the zeroth index causes the issue not at any other index (even at the 1st) (With version 3 of Virtuoso), with version 4 it causes issue even at index 1.  

Solution :- Added empty line at zeroth index to solve the issue

#### Screenshots

https://github.com/harness/harness-core-ui/assets/106532291/2e0a9018-06bc-4b10-afb9-ae9771949b6a

#### PR Checklist

<details>
<summary>Please check if your PR fulfils the following requirements</summary>

- [ ] Tests for the changes have been added. Ideally, include a test that fails without this PR but passes with it.
- [ ] Docs have been [added/updated](https://harness.atlassian.net/jira/software/c/projects/DOC/boards/40).
</details>

<details>
<summary>Use the following comments to re-trigger PR Checks</summary>

- Jest: `retrigger jest`
- Prettier: `retrigger prettier`
- Type Check: `retrigger typecheck`
- ESLint: `retrigger eslint`
- Sonar: `retrigger sonar`
- Standards: `retrigger standards`
- Build: `retrigger build`
- Title Check: `retrigger titlecheck`
- Feature Name Check: `trigger featurenamecheck`
- Coverage: `retrigger coverage`
- Rebase: `trigger rebase`
- Cypress Rest: `retrigger cypress-rest`
- Cypress CD: `retrigger cypress-cd`
- Cypress Pipeline: `retrigger cypress-pipeline`
- Cypress CV: `retrigger cypress-cv`
- Fix Prettier: `fix prettier`
- MessageMetadata: `trigger messagecheck`
- GitLeaks: `trigger gitLeaks`
</details>

## PR check failures and solutions
https://harness.atlassian.net/wiki/spaces/BT/pages/21106884744/PR+Checks+-+Failures+and+Solutions

### Rebase Instructions:
To rebase your branch onto the latest changes in the target branch, simply use the following command in a comment: `/rebase`

### CODE REVIEW

File: src/modules/70-pipeline/components/LogsContent/components/SingleSectionLogs.tsx [hunk 1]
```diff
@@ -24,7 +24,13 @@ export function SingleSectionLogs(
   const { state } = props
   const unitKey = state.logKeys[0]
   const unit = state.dataMap[unitKey]
-  const length = unit.data.length
+  const renderUnitData = React.useMemo(() => {
+    if (unit.data.length) {
```
Comment (sahithikolichala) [inline src/modules/70-pipeline/components/LogsContent/components/SingleSectionLogs.tsx:28]: may be add unit?.data?.length

File: src/modules/70-pipeline/components/LogsContent/components/SingleSectionLogs.tsx [hunk 2]
```diff
@@ -24,7 +24,13 @@ export function SingleSectionLogs(
...
+      return [{ text: { level: '', out: '', time: '' } }, ...unit.data]
+    }
+    return unit.data
+  }, [unit.data])
```
Comment (sahithikolichala) [inline src/modules/70-pipeline/components/LogsContent/components/SingleSectionLogs.tsx:32]: here too unit?.data

**Review by sahithikolichala — APPROVED** [review]

File: src/modules/70-pipeline/components/LogsContent/components/SingleSectionLogs.tsx (hunk 1 above)
Comment (vikrant-harness) [reply src/modules/70-pipeline/components/LogsContent/components/SingleSectionLogs.tsx:28]: These are not optional in the TS interface so not needed

File: src/modules/70-pipeline/components/LogsContent/components/SingleSectionLogs.tsx (hunk 2 above)
Comment (vikrant-harness) [reply src/modules/70-pipeline/components/LogsContent/components/SingleSectionLogs.tsx:32]: These are not optional in the TS interface so not needed

**Comment by vikrant-harness** [comment]
trigger messagecheck

**Review by abhinavrastogi-harness — APPROVED** [review]

**Review by PratyushGar — APPROVED** [review]

**Pull request merged**